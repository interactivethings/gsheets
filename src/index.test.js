require("isomorphic-fetch");
require("dotenv").config();
const getWorksheet = require("./index");

const TEST_SPREADSHEET_KEY = "1dmAQO0zCQz5SNUKalw9NNXwTM6TgDBZ820Ftw-cz5gU"; // new

test("Worksheet", () =>
  getWorksheet(TEST_SPREADSHEET_KEY, "something").then((worksheet) => {
    expect(worksheet).toMatchSnapshot();
  }));

test("Empty worksheet", () =>
  getWorksheet(TEST_SPREADSHEET_KEY, "nothing").then((worksheet) => {
    expect(worksheet).toMatchSnapshot();
  }));
