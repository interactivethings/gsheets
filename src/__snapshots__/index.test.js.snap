// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty worksheet 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Worksheet 1`] = `
Object {
  "data": Array [
    Object {
      "id": 1,
      "name": "foo",
      "should": "number id; string name; null value",
      "väl-ue": null,
    },
    Object {
      "id": 2,
      "name": " ",
      "should": "number id; string name (it's a single space!); number value",
      "väl-ue": 123456.789,
    },
    Object {
      "id": 3,
      "name": "baz",
      "should": "number id; string name; null value",
      "väl-ue": 0.000001,
    },
    Object {
      "id": 0,
      "name": "null",
      "should": "number id; string name; string value",
      "väl-ue": "1+2",
    },
    Object {
      "id": "4",
      "name": "money",
      "should": "string id; string name; numeric value",
      "väl-ue": 12.3,
    },
  ],
}
`;
